llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp

```diff
@@ -602,6 +602,8 @@ void OpenMPIRBuilder::finalize(Function *Fn) {
 
     Function *OuterFn = OI.getFunction();
     CodeExtractorAnalysisCache CEAC(*OuterFn);
+    unsigned AllocaAddressSpace =
+        Config.isTargetDevice() ? 0 : M.getDataLayout().getAllocaAddrSpace();
```

Comment:
Always use the DL?

Reply:
See comment: https://github.com/llvm/llvm-project/pull/66998#issuecomment-1729358182

Reply:
Fixed. I always use DL, and I perform address space casting if needed.